Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -238,6 +238,102 @@ std::string HiresTexture::GenBaseName(const u8* texture, size_t texture_size, co
 	return name;
 }
 
+static void PNGErrorFn(png_structp png_ptr, png_const_charp error_message)
+{
+	ERROR_LOG(VIDEO, "libpng error %s", error_message);
+	longjmp(png_ptr->jmpbuf, 1);
+}
+
+static void PNGWarnFn(png_structp png_ptr, png_const_charp error_message)
+{
+	WARN_LOG(VIDEO, "libpng warning %s", error_message);
+}
+
+static bool ReadPNG(File::IOFile *file, std::unique_ptr<u8[]> *output, size_t *out_data_size, u32 *out_width, u32 *out_height)
```

Comment:
And such helper functions should be moved to Common.

+1 if you replace the unique_ptr* with a callback for streaming into the backends:
std::function<std::pair<u8* /*data*/, int /*stride*/>(int /*width*/, int /*height*/)>